src/cross_module_fn.c

```diff
@@ -13,372 +13,96 @@
-}
-
-Datum
-ts_remove_reorder_policy(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->remove_reorder_policy(fcinfo));
-}
-
-Datum
-ts_remove_compress_chunks_policy(PG_FUNCTION_ARGS)
-{
-	return ts_cm_functions->remove_compress_chunks_policy(fcinfo);
-}
-
-Datum
-ts_alter_job_schedule(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->alter_job_schedule(fcinfo));
-}
-
-Datum
-ts_reorder_chunk(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->reorder_chunk(fcinfo));
-}
-
-Datum
-ts_move_chunk(PG_FUNCTION_ARGS)
-{
-	return ts_cm_functions->move_chunk(fcinfo);
-}
-
-Datum
-ts_continuous_agg_invalidation_trigger(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->continuous_agg_trigfn(fcinfo));
-}
-
-Datum
-ts_data_node_add(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->add_data_node(fcinfo));
-}
-
-Datum
-ts_data_node_delete(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->delete_data_node(fcinfo));
-}
-
-Datum
-ts_data_node_attach(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->attach_data_node(fcinfo));
-}
-
-Datum
-ts_data_node_ping(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->data_node_ping(fcinfo));
-}
-
-Datum
-ts_data_node_detach(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->detach_data_node(fcinfo));
-}
-
-Datum
-ts_data_node_block_new_chunks(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->data_node_block_new_chunks(fcinfo));
-}
-
-Datum
-ts_data_node_allow_new_chunks(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->data_node_allow_new_chunks(fcinfo));
-}
-
-Datum
-ts_chunk_set_default_data_node(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->set_chunk_default_data_node(fcinfo));
-}
-
-Datum
-ts_chunk_get_relstats(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->get_chunk_relstats(fcinfo));
-}
-
-Datum
-ts_chunk_get_colstats(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->get_chunk_colstats(fcinfo));
-}
-
-Datum
-ts_timescaledb_fdw_handler(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->timescaledb_fdw_handler(fcinfo));
-}
-
-Datum
-ts_timescaledb_fdw_validator(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->timescaledb_fdw_validator(fcinfo));
-}
-
-Datum
-ts_remote_txn_id_in(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->remote_txn_id_in(fcinfo));
-}
-
-Datum
-ts_remote_txn_id_out(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->remote_txn_id_out(fcinfo));
-}
-
-Datum
-ts_remote_txn_heal_data_node(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->remote_txn_heal_data_node(fcinfo));
-}
-
-Datum
-ts_remote_connection_cache_show(PG_FUNCTION_ARGS)
-{
-	PG_RETURN_DATUM(ts_cm_functions->remote_connection_cache_show(fcinfo));
-}
+#define TSL_FORWARD_CALL(func)                                                                     \
+	TS_FUNCTION_INFO_V1(ts_##func);                                                                \
+	Datum ts_##func(PG_FUNCTION_ARGS) { return ts_cm_functions->func(fcinfo); }
```

Comment:
`PG_RETURN_DATUM` does not do much, I know, but this is at least makes it clear that the function is expected to return a `Datum`
```suggestion
	Datum ts_##func(PG_FUNCTION_ARGS) { PG_RETURN_DATUM(ts_cm_functions->func(fcinfo)); }
```

Reply:
You could actually do something along these lines, if you want to be fancy and make something more generic:

```c
#define WRAPPER(FUNC, RETURNER) \
  TS_FUNCTION_INFO_V1(ts_##FUNC); \
  Datum ts_##FUNC (PG_FUNCTION_ARGS) { RETUNER(ts_cm_functions->func(fcinfo);
```
```